tools/toolchains/gcc.py

```diff
@@ -135,6 +137,19 @@ def __init__(self, target,  notify=None, macros=None, build_profile=None,
         self.use_distcc = (bool(getenv("DISTCC_POTENTIAL_HOSTS", False))
                            and not getenv("MBED_DISABLE_DISTCC", False))
 
+        self.coverage_cc = self.cc + ["--coverage", "-DENABLE_LIBGCOV_PORT"]
+        self.coverage_cppc = self.cppc + ["--coverage", "-DENABLE_LIBGCOV_PORT"]
+        self.coverage_ld = self.ld + ['--coverage', '-Wl,--wrap,GREENTEA_SETUP', '-Wl,--wrap,_Z25GREENTEA_TESTSUITE_RESULTi']
+
+        for flag in ["-DMBED_DEBUG"]:
```

Comment:
Could do with a comment here explaining what this is doing, and more importantly why.

Reply:
comments added